com.ibm.streamsx.inet/impl/java/src/com/ibm/streamsx/inet/http/HTTPPostOper.java

```diff
@@ -275,15 +311,26 @@ public synchronized void process(StreamingInput<Tuple> stream, Tuple tuple) thro
 		case TUPLE_FORM:
 		{
             Map<String, String> params = new HashMap<String, String>();
+            
             for (Attribute attribute : schema) {
-                params.put(attribute.getName(), tuple.getObject(attribute.getName()).toString());
+            	if (isAttributeToPost(attribute.getName())) {
+            		params.put(attribute.getName(), tuple.getObject(attribute.getName()).toString());
+            	}
             }
-            req.setParams(params);		    
+            req.setParams(params);	
+            break;
 		}
 		case TUPLE_JSON:
 		{
             JSONEncoding<JSONObject, JSONArray> je = EncodingFactory.getJSONEncoding();
-            req.setParams(je.encodeAsString(tuple));          
+            JSONObject jo = je.encodeTuple(tuple);
+            
+            for (Iterator<String> it = jo.keySet().iterator(); it.hasNext();) {
```

Comment:
If I switch to 
	    `
Set<String> keys =jo.keySet();
	    for (String attrName : keys) {
	    for (String attrName : keys) {
            	if (!isAttributeToPost(attrName)) {
		    jo.remove(attrName);
           }`


It compiles fine but I get a runtime error, 'java.util.ConcurrentModificationException', which
surprized me since I thought